kolibri/utils/main.py

```diff
@@ -0,0 +1,265 @@
+                "please immediately back up all files in the .kolibri folder that"
+                "end in .sqlite3, .sqlite3-shm, .sqlite3-wal, or .log and then"
+                "contact Learning Equality. Thank you!"
+            )
+        raise
+
+
+def initialize(
+    skip_update=False,
+    settings=None,
+    debug=False,
+    debug_database=False,
+    no_input=False,
+    pythonpath=None,
+):  # noqa: max-complexity=12
+    """
+    This should be called before starting the Kolibri app, it initializes Kolibri plugins
+    and sets up Django.
+    """
+    check_debian_user(no_input)
+
+    setup_logging(debug=debug, debug_database=debug_database)
+
+    default_options = DefaultDjangoOptions(settings, pythonpath)
+
+    handle_default_options(default_options)
+
+    # Do this here so that we can fix any issues with our configuration file before
+    # we attempt to set up django.
+    autoremove_unavailable_plugins()
+
+    # Check if there is an options.ini file exist inside the KOLIBRI_HOME folder
+    check_default_options_exist()
+
+    version = get_version()
+
+    updated = version_updated(kolibri.__version__, version)
+
+    if updated:
+        check_plugin_config_file_location(version)
+        # Reset the enabled plugins to the defaults
+        # This needs to be run before dbbackup because
+        # dbbackup relies on settings.INSTALLED_APPS
+        enable_new_default_plugins()
+
+    _setup_django()
+
+    if updated and not skip_update:
+        conditional_backup(kolibri.__version__, version)
+
+        if version:
+            logger.info(
+                "Version was {old}, new version: {new}".format(
+                    old=version, new=kolibri.__version__
+                )
+            )
+        else:
+            logger.info("New install, version: {new}".format(new=kolibri.__version__))
+        update(version, kolibri.__version__)
+
+    check_content_directory_exists_and_writable()
+
+    if not skip_update:
+        # Run any plugin specific updates here in case they were missed by
+        # our Kolibri version based update logic.
+        run_plugin_updates()
+
+        check_django_stack_ready()
+
+        try:
+            check_database_is_migrated()
+        except DatabaseNotMigrated:
+            try:
+                _migrate_databases()
+            except Exception as e:
+                logging.error(
+                    "The database was not fully migrated. Tried to "
+                    "migrate the database and an error occurred: "
+                    "{}".format(e)
+                )
+                raise
+        except DatabaseInaccessible as e:
+            logging.error(
+                "Tried to check that the database was accessible "
+                "and an error occurred: {}".format(e)
+            )
+            raise
+
+
+def update(old_version, new_version):
+    """
+    Called whenever a version change in kolibri is detected
+    """
```

Comment:
In the previous version of this function, the comment had also this text:

> TODO: We should look at version numbers of external plugins, too!

is there a reason to remove it now? I don't see any differences affecting the plugins

Reply:
No, except that it was addressed a long time ago, and I only just noticed that it is still here!